src/SoftGlobalScope.jl

```diff
@@ -152,8 +152,21 @@ else
             return Expr(ex.head, (_softscope.(ex.args, Ref(globals), Ref(locals), insertglobal))...)
         elseif isexpr(ex, :kw)
             return Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal))
-        elseif insertglobal && ex.head in assignments && ex.args[1] in globals && !(ex.args[1] in locals)
-            return Expr(:global, Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+        elseif insertglobal && ex.head in assignments 
+            if ex.args[1] in globals && !(ex.args[1] in locals)
+                # Simple assignment
+                return Expr(:global, Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+            elseif isexpr(ex.args[1], :tuple)
+                # Assignment to a tuple
+                vars = [var for var in localvars(ex.args[1].args) if (var in globals) && !(var in locals)]
+                if length(vars) > 0
+                    return Expr(:block, Expr(:global, vars...), Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal)))
+                else
+                    return Expr(ex.head, ex.args[1], _softscope(ex.args[2], globals, locals, insertglobal))
+                end
+            else
+                return ex
```

Comment:
Also `Expr(ex.head, _softscope.(ex.args, globals, locals, insertglobal)...)`?  Updated.

Reply:
That needs a few `Ref`s to allow broadcasting properly (added). 

Also it breaks `let; f(a) = (a = 1); end` since it softscopes the function definition when it shouldn't. That's easy enough to fix (done). 

Also, softscope needs to understand `:ref`s to do `a[global b += 1] = 1` (also done). 

However, getting your test case to work properly is more difficult since it requires softscoping a tuple which isn't a simple recursive fix since assignment within a tuple can create a named tuple under the right circumstances. It's probably not that hard but I'm out of time at the moment. (Personally I'd ignore this case as it's not a particularly common use case as far as I'm aware.)